### ISSUE TITLE
performance issue with nginx plugin

### ISSUE THREAD

**Issue by cameronbraid** [vtex/faststore#731]
Hi

I have identified that the nginx plugin is taking 7 minutes to run for my production build.

```
success write out nginx configuration - 418.332s
```

which for a total (incremental) build time of 12.5 minutes seems a bit long :)

```
info Done building in 746.185893502 sec
```

My config is pretty simple 

```
    {
      resolve: '@vtex/gatsby-plugin-nginx',
      options: {
        // disableBrotliEncoding: true,
        transformHeaders: (headers : string[], path: string) : string[] => {

          let re = /^\/[^/]*-[0-9A-Fa-f]{20}\.js(\.map)?(\.LICENSE\.txt)?$/
          if (re.test(path)) {
            return [...headers.filter(h=>!h.includes("Cache-Control")), `Cache-Control: public, max-age=31536000, immutable`]
          }

          return headers
        }
      },
    },

```

Though I do have a few pages - **1347** in total

Are there any specific pointers on where I should start looking to find performance hot spots ?

**Comment by cameronbraid** [comment]
Oh, I should also say version `0.353.0`

**Comment by cameronbraid** [comment]
Some quickly hacked profiling shows 99.9% of the time is in generateNginxConfiguration

**Comment by tlgimenes** [comment]
Thanks for signaling this issue @cameronbraid!  

Checking the stats included in #727, our CI reports max of 11s. However, it looks to me that some sites take longer than others. Maybe you are falling on the worst case scenario.

I don't think we can prioritize this issue right now. However, we are open for pull requests if you feel interested to. Thanks again for contributing!

**Comment by cameronbraid** [comment]
Fixed https://github.com/vtex/faststore/pull/737
